src/Angor/Shared/DerivationOperations.cs

```diff
@@ -216,6 +216,22 @@ public Key DeriveProjectNostrPrivateKey(WalletWords walletWords, int index)
 
         return extKey.PrivateKey;
     }
+    
+    public async Task<Key> DeriveProjectNostrPrivateKeyAsync(WalletWords walletWords, int index)
+    {
+        // founder key is derived from the path m/5'
+        ExtKey extendedKey = GetExtendedKey(walletWords);
+
+        var path = $"m/44'/1237'/{index}/0/0";
+
+        var task = Task.Run(() => extendedKey.Derive(new KeyPath(path)));
```

Comment:
this doesn't help in anyway to make it faster, drop the Task here if you ask me it is just confusing.

Reply:
Not faster but the UI doesn't freeze when it runs in a task

Reply:
But later down you await the task is it means the browser will still freeze no?